platform/darwin/src/MGLStyle.mm

```diff
@@ -293,7 +293,7 @@ - (void)insertObject:(MGLStyleLayer *)styleLayer inLayersAtIndex:(NSUInteger)ind
         }
```

Comment:
What if you pass in an index of 0 and the number of layers is 0? The layer should be inserted and no exception should be raised. `-[NSMutableArray insertObject:atIndex:]` allows index=size but not + 1. The second argument still holds but the first should be `>`.